### PR TITLE
KOGITO-4167 Set optaplanner before apps

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-4167

Related PRs:
- https://github.com/kiegroup/kogito-runtimes/pull/976
- https://github.com/kiegroup/kogito-apps/pull/589
- https://github.com/kiegroup/kogito-examples/pull/514

Test summary
- kogito-apps-sonarcloud => OK

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -23,40 +25,74 @@ pipeline {
...
             }
         }
-        stage('Build kogito-apps') {
+        stage('Build OptaPlanner') {
             steps {
-                mavenCleanInstall("kogito-apps", false, ["run-code-coverage"])
-                runMaven("validate", "kogito-apps", false, ["sonarcloud-analysis"], "-e -nsu")
+                script {
+                    // Skip unnecessary plugins to save time.
+                    getMavenCommand('optaplanner')
```
Comment (rsynek) [inline Jenkinsfile:53]: Is it important to run also tests (I don't see them being skipped)? In the 7.x world, the "upstream" repositories were built without running tests.

For the purpose of running just the compilation (without tests and unnecessary plugins), we've recently introduced the `quickProfile`: https://github.com/kiegroup/optaplanner/blob/master/build/optaplanner-build-parent/pom.xml#L1094

Thus, `mvn clean install -Dquickly` would be sufficient and hides the knowledge of which plugins should be skipped.

**Review by rsynek — APPROVED** [review]
1 suggestion, generally looks good.

File: Jenkinsfile (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile:53]: that is a cool one ! gonna use it. I guess we could introduce something similar (quick profile) into Kogito 
cc @evacchi @danielezonca wdyt ?

Comment (rsynek) [reply Jenkinsfile:53]: Inspiration taken from the Quarkus project :)

Comment (danielezonca) [reply Jenkinsfile:53]: @radtriste 
Fine for me, not sure how many step we can remove but it is in general something good

**Review by danielezonca — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile:53]: opened https://issues.redhat.com/browse/KOGITO-4183

**Pull request merged**